zkevm-circuits/src/evm_circuit.rs

```diff
@@ -332,11 +332,17 @@ impl<F: Field> SubCircuit<F> for EvmCircuit<F> {
     ) -> Result<(), Error> {
         let block = self.block.as_ref().unwrap();
 
+        let tx_max_challenge_pow: Option<usize> = if block.txs.len() > 0 {
+            Some(block.txs.iter().map(|tx| tx.rlp_signed.len()).max().unwrap())
+        } else {
+            None
+        };
+
```

Comment:
It is wrong to set `tx_max_challenge_pow` like this. For when we are running pk/vk generation, the block is empty (i.e. `block.txs.is_empty()`). To fix this, I suggest that you just set `4096*31 - 62` as the maximum.